zetaclient/chains/solana/observer/inbound.go

```diff
@@ -315,12 +315,50 @@ func (ob *Observer) ParseInboundAsDeposit(
 // ParseInboundAsDepositSPL tries to parse an instruction as a 'deposit_spl_token'.
 // It returns nil if the instruction can't be parsed as a 'deposit_spl_token'.
 func (ob *Observer) ParseInboundAsDepositSPL(
-	_ *solana.Transaction,
-	_ int,
-	_ uint64,
+	tx *solana.Transaction,
+	instructionIndex int,
+	slot uint64,
 ) (*clienttypes.InboundEvent, error) {
-	// not implemented yet
-	return nil, nil
+	// get instruction by index
+	instruction := tx.Message.Instructions[instructionIndex]
+
+	// try deserializing instruction as a 'deposit_spl_token'
+	var inst solanacontracts.DepositSPLInstructionParams
+	err := borsh.Deserialize(&inst, instruction.Data)
+	if err != nil {
+		return nil, nil
```

Comment:
- This logic should be implemented within solana contracts package. See TON example: https://github.com/zeta-chain/node/blob/041df59fe4d012ecd293a0bb530e1c25dae9a381/zetaclient/chains/ton/observer/inbound.go#L109 
- Let's avoid returning `(nil, nil)`. Instead, you can create a sentinel error e.g. `ErrParse`